pkgs/by-name/ni/nixos-init/README.md

```diff
@@ -0,0 +1,66 @@
+# nixos-init
+
+A system for the initialization of NixOS.
+
+The most important task of nixos-init is to work around the constraints of the
+Filesystem Hierarchy Standard (FHS) that are imposed by other tools (most
+importantly systemd itself).
+
+The primary design principle is to do the minimal work requird to start
+systemd. Everything that can be done later SHOULD be done later (i.e. after
+systemd has already started). This isn't controversial either, this is the
+basic principle behind initrds in the first place.
+
+Extending this init should be done with care and only when strictly necessary.
+Extending it should always be a last resort. It is explitly not designed to be
+extended for third party extensions. I would go so far that all third party
+extensions should be forbidded.
+
+The goal of nixos-init is to eventually entriely replace `activationScripts`
+for booting, enable bashless activation and ultimately make NixOS overall more
+robust.
+
+## Reasoning
+
+- We already have a native API that can be used to easily extend the system
+  (`systemd.services`). This is in all ways superior to the stringified and
+  sequential nature of the `activationScripts`.
+- Most things can be started much later than you would typically assume.
+  Because systemd services are parallelized, this should improve startup time.
+- Using a fully fledged programming language makes writing correct software
+  easier and should improve the quality of the NixOS boot code.
```

Comment:
I'd move this one bullet point up, and maybe add a "For the remaining functionality needed, …" in front.